### PR TITLE
add query context as argument to node identification # object from id

### PR DESCRIPTION
**Pull request by charlieschwabacher**
I ended up w/ a Session type which includes the current user as a field (so that user can be logged in and out using mutations), I needed the context here in order to build a Session object.

I am not 100% sure I have this set up in an ideal way - do you have any objections to including context as an argument here?

### CODE REVIEW

**Comment by rmosolgo** [comment]
It seems a bit weird because it allows people to break the rule of _one UUID_ -> _one object_ (because you could theoretically use something _other_ than the UUID to find an object). 

But I hear your point that if the person requests some info that's in the session, you need _some_ way to return that info. 

I'm on the road this week so I can't totally think it through right now, let me get back to it on Monday!

**Comment by staugaard** [comment]
In a multi tenant world, you always want to scope your database queries to the current account. Passing in the current account into the context, I think is the natural way to do it. So having access to context everywhere you need to do database operations is crucial.

:+1:

**Comment by staugaard** [comment]
@rmosolgo any chance we can get this merged?

**Comment by cesarandreu** [comment]
@rmosolgo

I think this is the right approach. This is the same thing that graphql-relay-js does with [`nodeDefinitions`](https://github.com/graphql/graphql-relay-js/blob/master/src/node/node.js#L46-L76) as well. 

:+1:

**Comment by rmosolgo** [comment]
Thanks for that pointer to the JS implementation -- that makes it clear that we should have this too! I'll merge it this weekend. It's the first breaking change though. I guess it's time to add a changelog!

File: lib/graphql/relay/base_connection.rb [hunk 1]
```diff
@@ -36,8 +36,8 @@ def self.create_type(wrapped_type, &block)
       # @param [Object] A collection of items (eg, Array, AR::Relation)
       # @return [subclass of BaseConnection] a connection Class for wrapping `items`
       def self.connection_for_items(items)
-        implementation = CONNECTION_IMPLEMENTATIONS.find do |items_class, connection_class|
-          items.is_a?(items_class)
+        implementation = CONNECTION_IMPLEMENTATIONS.find do |items_class_name, connection_class|
```
Comment (rmosolgo) [inline lib/graphql/relay/base_connection.rb]: Why is it better to use `class.name` instead of `class` as keys for this hash?

Comment (staugaard) [inline lib/graphql/relay/base_connection.rb]: If you hold a reference to the class then you are preventing rails class reloading in development. That's all I can think of.

Comment (rmosolgo) [inline lib/graphql/relay/base_connection.rb]: Hmmm I wonder. Whatever the case, it would be suitable for a separate PR with its own description etc 

I'll give it a couple days to see if @charlieschwabacher wants to clean up this branch, otherwise I'll just grab the previous commits and and merge them in!

Comment (staugaard) [inline lib/graphql/relay/base_connection.rb]: Agreed. This should not be part of this PR.

Comment (charlieschwabacher) [inline lib/graphql/relay/base_connection.rb]: Hey - sorry, didn't mean to add this here - I just have the Gemfile of my project pointing at that branch and forgotten that I had this pull open.

That change was because every time I modified the items class, it would be loaded by rails as a new object and I would have to restart the server - I'll open a separate pull w/ that.

**Comment by rmosolgo** [comment]
:confetti_ball:  thanks so much, released in 0.6.0!
